### ISSUE TITLE
Constructor does not use instance argument for type inference

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#2350]
Another bug in the series `isHidden`, see #2347, #2348, #2349.
```agda
postulate A : Set

data D {{a : A}} : Set where
  c : D

test : {{a b : A}} → D
test {{a}} = c {{a}}
```
Agda complains about not being able to infer instance for `D`, since it ignores it in the constructor.

**Comment by andreasabel** [comment]
Even worse, Agda may ignore a given instance parameter of a constructor:
It accepts this:
```agda
data D {{a}} (A : Set a) : Set a where
  c : A → D A

test : ∀ ℓ {ℓ'} (A : Set ℓ') {B : Set ℓ} (a : A) → D A
test ℓ A a = c {{ℓ}} a
```
The level `ℓ` is the wrong one, but it is ignored by Agda.

**Comment by asr** [comment]
I cherry-picked the fix into release-2.5.2.

**Comment by andreasabel** [comment]
Great, thanks!
